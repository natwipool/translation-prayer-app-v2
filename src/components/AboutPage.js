import React from 'react';

const AboutPage = () => (
  <div>
    <div className="page-header">
      <div className="content-container-body">
        <h2 className="page-header__title">เกี่ยวกับ</h2>
      </div>
    </div>
    <div className="content-container-body">
      <img className="cover-image" src="/images/about.jpg" />
      <div className="text-container">
        <p>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Praesent
          tempus nisl odio, in sollicitudin nunc lacinia et. Sed molestie, nibh
          consectetur pulvinar pulvinar, urna ligula mattis tellus, quis
          molestie orci lectus eget libero. Curabitur ac ex turpis. Curabitur
          ante nunc, semper sed imperdiet quis, vehicula eget eros. Fusce id
          fermentum dolor. Ut gravida mauris enim, non placerat eros blandit
          nec. Nulla malesuada urna ut felis maximus, at luctus lectus
          hendrerit. Interdum et malesuada fames ac ante ipsum primis in
          faucibus. Pellentesque ultricies quam quis elit ultricies, ut ultrices
          sapien fermentum.
        </p>
        <p>
          Icons made by{' '}
          <a
            href="https://www.flaticon.com/authors/zlatko-najdenovski"
            title="Zlatko Najdenovski"
          >
            Zlatko Najdenovski
          </a>{' '}
          from{' '}
          <a href="https://www.flaticon.com/" title="Flaticon">
            www.flaticon.com
          </a>{' '}
          is licensed by{' '}
          <a
            href="http://creativecommons.org/licenses/by/3.0/"
            title="Creative Commons BY 3.0"
            target="_blank"
          >
            CC 3.0 BY
          </a>
        </p>
      </div>
    </div>
  </div>
);

export default AboutPage;
