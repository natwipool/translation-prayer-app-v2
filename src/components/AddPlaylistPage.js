import React from 'react';
import { connect } from 'react-redux';
import PlaylistForm from './PlaylistForm';
import { addPlaylist } from '../actions/playlists';

export class AddPlaylistPage extends React.Component {
  onSubmit = playlist => {
    this.props.addPlaylist(playlist);
    this.props.history.push('/playlists');
  };
  render() {
    return (
      <div>
        <h3>สร้างรายการสวดมนต์</h3>
        <PlaylistForm onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addPlaylist: playlist => dispatch(addPlaylist(playlist))
});

export default connect(undefined, mapDispatchToProps)(AddPlaylistPage);
