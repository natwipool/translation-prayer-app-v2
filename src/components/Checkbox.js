import React from 'react';

const Checkbox = props => {
  const onCheckboxChange = e => {
    const list = e.target.value;

    if (e.target.checked) {
      props.onAddLists(list);
    } else if (!e.target.checked) {
      props.onRemoveLists(list);
    }
  };

  return (
    <div>
      <label htmlFor={props.precept} className="checkbox">
        <input
          type="checkbox"
          id={props.precept}
          checked={props.lists.find(list => list === props.precept)}
          value={props.precept}
          onChange={onCheckboxChange}
        />
        <div className="checkbox-content">
        <span />
          
            <p className="checkbox-content__title">{props.precept}</p>
            
            {
              props.lists.indexOf(props.precept) > -1 &&
              <p className="checkbox-content__number">{props.lists.indexOf(props.precept) + 1}</p>
            } 
          
        </div>
      </label>
    </div>
  );
};

export default Checkbox;
