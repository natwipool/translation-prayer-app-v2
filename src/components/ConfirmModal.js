import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import { startRemovePlaylist } from '../actions/playlists';

export class ConfirmModal extends React.Component {
  onRemove = () => {
    this.props.startRemovePlaylist({ id: this.props.playlist.id });
    this.props.history.push('/playlists');
  };
  render() {
    return (
      <Modal
        isOpen={!!this.props.showConfirmModal}
        contentLabel="confirm-delete"
        onRequestClose={this.props.handleCloseModal}
        ariaHideApp={false}
        closeTimeoutMS={200}
        className="Modal"
      >
        <div className="confirm-modal-content">
          <h3 className="modal-confirm__title">ลบรายการ ?</h3>
          <h3 className="modal-confirm__title">
            {this.props.playlist.description}
          </h3>
        <div className="confirm-modal-group-button">
          <button
            className="button--secondary del-button"
            onClick={this.onRemove}
          >
            ตกลง
          </button>
          <button
            className="button--secondary edit-button"
            onClick={this.props.handleCloseModal}
          >
            ยกเลิก
          </button>
          </div>
        </div>
      </Modal>
    );
  }
}

const mapDispatchToProps = (dispatch, props) => ({
  startRemovePlaylist: id => dispatch(startRemovePlaylist(id))
});

export default connect(undefined, mapDispatchToProps)(ConfirmModal);
