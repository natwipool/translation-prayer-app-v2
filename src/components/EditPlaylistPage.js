import React from 'react';
import { connect } from 'react-redux';
import PlaylistForm from './PlaylistForm';
import { startEditPlaylist } from '../actions/playlists';

export class EditPlaylistPage extends React.Component {
  onSubmit = playlist => {
    this.props.startEditPlaylist(this.props.playlist.id, playlist);
    this.props.history.push(`/playlist/${this.props.playlist.id}`);
  };
  render() {
    return (
      <div>
        <h3>แก้ไขรายการสวดมนต์</h3>
        <PlaylistForm playlist={this.props.playlist} onSubmit={this.onSubmit} />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  playlist: state.playlists.find(
    playlist => playlist.id === props.match.params.id
  )
});

const mapDispatchToProps = dispatch => ({
  startEditPlaylist: (id, playlist) => dispatch(startEditPlaylist(id, playlist))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditPlaylistPage);
