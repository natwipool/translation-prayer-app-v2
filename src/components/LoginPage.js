import React from 'react';
import { connect } from 'react-redux';
import { startLoginWithGoogle, startLoginWithFacebook } from '../actions/auth';

export const LoginPage = ({ startLoginWithGoogle, startLoginWithFacebook }) => (
  <div>
    <button onClick={startLoginWithGoogle}>Login with Google</button>
    <button onClick={startLoginWithFacebook}>Login with Facebook</button>
  </div>
);

const mapDispatchToProps = (dispatch) => ({
  startLoginWithGoogle: () => dispatch(startLoginWithGoogle()),
  startLoginWithFacebook: () => dispatch(startLoginWithFacebook())
})

export default connect(undefined, mapDispatchToProps)(LoginPage);