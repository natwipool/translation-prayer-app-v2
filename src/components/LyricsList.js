import React from 'react';
import LyricsListItem from './LyricsListItem';

const LyricsList = (props) => (
  <div>
    {props.playlist.lyrics.map((lyric, index) => 
      <LyricsListItem
        key={index}
        lyric={lyric}
      />
    )}
  </div>
);

export default LyricsList;