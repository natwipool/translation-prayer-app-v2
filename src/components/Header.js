import React from 'react';
import { connect } from 'react-redux';
import { Link, NavLink } from 'react-router-dom';
import { Navbar, Nav, NavItem } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';
import LoginPage from '../components/LoginPage';
import { startLogout } from '../actions/auth';
import { handleShowLoginModal } from '../actions/modal';

export const Header = props => (
  <header>
    <Navbar inverse collapseOnSelect>
      <div className="content-container">
        <Navbar.Header>
          <Navbar.Brand>
            <Link to="/">สวดมนต์แปล</Link>
          </Navbar.Brand>
          <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
          <Nav>
            <LinkContainer to="/trans-prayers">
              <NavItem>สารบัญ</NavItem>
            </LinkContainer>
            <LinkContainer to="/playlists">
              <NavItem>บทสวดมนต์ของฉัน</NavItem>
            </LinkContainer>
            <LinkContainer to="/about">
              <NavItem>เกี่ยวกับ</NavItem>
            </LinkContainer>
          </Nav>
          <Nav pullRight>
            <NavItem>
              {!props.isAuthenticated ? (
                <button 
                  className="button header-button"
                  onClick={() => {
                    props.handleShowLoginModal(true);
                  }}
                >
                  เข้าสู่ระบบ
                </button>
              ) : (
                <button 
                  className="button header-button logout-button"
                  onClick={props.startLogout}
                >
                  ออกจากระบบ
                </button>
              )}
            </NavItem>
          </Nav>
        </Navbar.Collapse>
      </div>
    </Navbar>
    <LoginPage
      showLoginModal={!!props.loginModal}
      handleHideLoginModal={() => {
        props.handleShowLoginModal();
      }}
    />
  </header>
);

const mapStateToProps = state => ({
  isAuthenticated: !!state.auth.uid,
  loginModal: state.modal.showLoginModal
});

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout()),
  handleShowLoginModal: isShow => dispatch(handleShowLoginModal(isShow))
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);
