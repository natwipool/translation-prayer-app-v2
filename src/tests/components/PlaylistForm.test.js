import React from 'react';
import { shallow } from 'enzyme';
import { PlaylistForm } from '../../components/PlaylistForm';
import transPrayers from '../fixtures/transPrayers';
import playlists from '../fixtures/playlists';

test('should render PlaylistForm correctly', () => {
  const wrapper = shallow(<PlaylistForm transPrayers={transPrayers}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should render PlaylistForm with playlists data', () => {
  const wrapper = shallow(<PlaylistForm
    transPrayers={transPrayers}
    playlist={playlists[2]}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('should render error for invalid form submission', () => {
  const wrapper = shallow(<PlaylistForm transPrayers={transPrayers}/>);
  expect(wrapper).toMatchSnapshot();
  wrapper.find('form').simulate('submit', {
    preventDefault: () => { }
  });
  expect(wrapper.state('error').length).toBeGreaterThan(0);
  expect(wrapper).toMatchSnapshot();
});

test('should set description on input change', () => {
  const value = 'New description';
  const wrapper = shallow(<PlaylistForm transPrayers={transPrayers}/>);
  wrapper.find('input').simulate('change', {
    target: { value }
  });
  expect(wrapper.state('description')).toBe(value)
});

test('should call onAddlists on input change', () => {
  const wrapper = shallow(<PlaylistForm transPrayers={transPrayers}/>);
  wrapper.find('Checkbox').at(0).prop('onAddLists')('123');
  expect(wrapper.state('lists').length).toBe(1);
});

test('should call onRemoveLists on input change', () => {
  const wrapper = shallow(<PlaylistForm
    transPrayers={transPrayers}
    playlist={playlists[0]}
  />);
  wrapper.find('Checkbox').at(0).prop('onRemoveLists')('คำทำวัตรเช้า');
  expect(wrapper.state('lists')).toEqual(['ตังขณิกปัจจเวกขณปาฐะ', 'บทพิจารณาสังขาร']);
});

test('should call onSubmit prop for valid form submission', () => {
  const onSubmitSpy = jest.fn();
  const wrapper = shallow(<PlaylistForm
    transPrayers={transPrayers}
    playlist={playlists[1]}
    onSubmit={onSubmitSpy}
  />);
  wrapper.find('form').simulate('submit', {
    preventDefault: () => { }
  });
  expect(wrapper.state('error')).toBe('');
  expect(onSubmitSpy).toHaveBeenLastCalledWith({
    description: playlists[1].description,
    lists: playlists[1].lists
  });
});

