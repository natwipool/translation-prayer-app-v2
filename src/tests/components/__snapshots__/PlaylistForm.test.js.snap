// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PlaylistForm correctly 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    onChange={[Function]}
    placeholder="ชื่อรายการ"
    type="text"
    value=""
  />
  <Connect(PlaylistSummary)
    playlist={
      Object {
        "description": "",
        "error": "",
        "lists": Array [],
      }
    }
  />
  <Checkbox
    category="บทสวดมนต์ภาคเย็น"
    duration={155.99425}
    filename="11_เขมาเขมสรณคมนปริทีปิกาคาถา.mp3"
    id="11"
    key="0"
    lists={Array []}
    lyrics={
      Array [
        "abc",
        "dfg",
      ]
    }
    onAddLists={[Function]}
    onRemoveLists={[Function]}
    precept="เขมาเขมสรณคมนปริทีปิกาคาถา"
  />
  <Checkbox
    category="บทสวดมนต์ภาคเย็น"
    duration={99.99425}
    filename="12_อริยธนคาถา.mp3"
    id="12"
    key="1"
    lists={Array []}
    lyrics={
      Array [
        "123",
        "456",
      ]
    }
    onAddLists={[Function]}
    onRemoveLists={[Function]}
    precept="อริยธนคาถา"
  />
  <Checkbox
    category="บทสวดมนต์ภาคเย็น"
    duration={212.99425}
    filename="13_ติลักขณาทิคาถา.mp3"
    id="13"
    key="2"
    lists={Array []}
    lyrics={
      Array [
        "789",
        "123",
      ]
    }
    onAddLists={[Function]}
    onRemoveLists={[Function]}
    precept="ติลักขณาทิคาถา"
  />
  <div>
    <button
      className="button form-button"
    >
      เพิ่มรายการ
    </button>
  </div>
</form>
`;

exports[`should render PlaylistForm with playlists data 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    onChange={[Function]}
    placeholder="ชื่อรายการ"
    type="text"
    value="Playlist name three"
  />
  <Connect(PlaylistSummary)
    playlist={
      Object {
        "description": "Playlist name three",
        "error": "",
        "lists": Array [
          "อานาปานสติสูตร",
          "อริยมรรคมีองค์แปด",
          "ธัมมจักกัปปวัตตนสุตตปาฐะ",
        ],
      }
    }
  />
  <Checkbox
    category="บทสวดมนต์ภาคเย็น"
    duration={155.99425}
    filename="11_เขมาเขมสรณคมนปริทีปิกาคาถา.mp3"
    id="11"
    key="0"
    lists={
      Array [
        "อานาปานสติสูตร",
        "อริยมรรคมีองค์แปด",
        "ธัมมจักกัปปวัตตนสุตตปาฐะ",
      ]
    }
    lyrics={
      Array [
        "abc",
        "dfg",
      ]
    }
    onAddLists={[Function]}
    onRemoveLists={[Function]}
    precept="เขมาเขมสรณคมนปริทีปิกาคาถา"
  />
  <Checkbox
    category="บทสวดมนต์ภาคเย็น"
    duration={99.99425}
    filename="12_อริยธนคาถา.mp3"
    id="12"
    key="1"
    lists={
      Array [
        "อานาปานสติสูตร",
        "อริยมรรคมีองค์แปด",
        "ธัมมจักกัปปวัตตนสุตตปาฐะ",
      ]
    }
    lyrics={
      Array [
        "123",
        "456",
      ]
    }
    onAddLists={[Function]}
    onRemoveLists={[Function]}
    precept="อริยธนคาถา"
  />
  <Checkbox
    category="บทสวดมนต์ภาคเย็น"
    duration={212.99425}
    filename="13_ติลักขณาทิคาถา.mp3"
    id="13"
    key="2"
    lists={
      Array [
        "อานาปานสติสูตร",
        "อริยมรรคมีองค์แปด",
        "ธัมมจักกัปปวัตตนสุตตปาฐะ",
      ]
    }
    lyrics={
      Array [
        "789",
        "123",
      ]
    }
    onAddLists={[Function]}
    onRemoveLists={[Function]}
    precept="ติลักขณาทิคาถา"
  />
  <div>
    <button
      className="button form-button"
    >
      เพิ่มรายการ
    </button>
  </div>
</form>
`;

exports[`should render error for invalid form submission 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    onChange={[Function]}
    placeholder="ชื่อรายการ"
    type="text"
    value=""
  />
  <Connect(PlaylistSummary)
    playlist={
      Object {
        "description": "",
        "error": "",
        "lists": Array [],
      }
    }
  />
  <Checkbox
    category="บทสวดมนต์ภาคเย็น"
    duration={155.99425}
    filename="11_เขมาเขมสรณคมนปริทีปิกาคาถา.mp3"
    id="11"
    key="0"
    lists={Array []}
    lyrics={
      Array [
        "abc",
        "dfg",
      ]
    }
    onAddLists={[Function]}
    onRemoveLists={[Function]}
    precept="เขมาเขมสรณคมนปริทีปิกาคาถา"
  />
  <Checkbox
    category="บทสวดมนต์ภาคเย็น"
    duration={99.99425}
    filename="12_อริยธนคาถา.mp3"
    id="12"
    key="1"
    lists={Array []}
    lyrics={
      Array [
        "123",
        "456",
      ]
    }
    onAddLists={[Function]}
    onRemoveLists={[Function]}
    precept="อริยธนคาถา"
  />
  <Checkbox
    category="บทสวดมนต์ภาคเย็น"
    duration={212.99425}
    filename="13_ติลักขณาทิคาถา.mp3"
    id="13"
    key="2"
    lists={Array []}
    lyrics={
      Array [
        "789",
        "123",
      ]
    }
    onAddLists={[Function]}
    onRemoveLists={[Function]}
    precept="ติลักขณาทิคาถา"
  />
  <div>
    <button
      className="button form-button"
    >
      เพิ่มรายการ
    </button>
  </div>
</form>
`;

exports[`should render error for invalid form submission 2`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <p
    className="form__error"
  >
    กรุณาใส่ชื่อรายการและเลือกบทสวดมนต์
  </p>
  <input
    autoFocus={true}
    onChange={[Function]}
    placeholder="ชื่อรายการ"
    type="text"
    value=""
  />
  <Connect(PlaylistSummary)
    playlist={
      Object {
        "description": "",
        "error": "กรุณาใส่ชื่อรายการและเลือกบทสวดมนต์",
        "lists": Array [],
      }
    }
  />
  <Checkbox
    category="บทสวดมนต์ภาคเย็น"
    duration={155.99425}
    filename="11_เขมาเขมสรณคมนปริทีปิกาคาถา.mp3"
    id="11"
    key="0"
    lists={Array []}
    lyrics={
      Array [
        "abc",
        "dfg",
      ]
    }
    onAddLists={[Function]}
    onRemoveLists={[Function]}
    precept="เขมาเขมสรณคมนปริทีปิกาคาถา"
  />
  <Checkbox
    category="บทสวดมนต์ภาคเย็น"
    duration={99.99425}
    filename="12_อริยธนคาถา.mp3"
    id="12"
    key="1"
    lists={Array []}
    lyrics={
      Array [
        "123",
        "456",
      ]
    }
    onAddLists={[Function]}
    onRemoveLists={[Function]}
    precept="อริยธนคาถา"
  />
  <Checkbox
    category="บทสวดมนต์ภาคเย็น"
    duration={212.99425}
    filename="13_ติลักขณาทิคาถา.mp3"
    id="13"
    key="2"
    lists={Array []}
    lyrics={
      Array [
        "789",
        "123",
      ]
    }
    onAddLists={[Function]}
    onRemoveLists={[Function]}
    precept="ติลักขณาทิคาถา"
  />
  <div>
    <button
      className="button form-button"
    >
      เพิ่มรายการ
    </button>
  </div>
</form>
`;
